### PR TITLE
Using of memory allocation

### PR DESCRIPTION
**Pull request by TimurKM**
Please, review

### CODE REVIEW

File: TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+int main(int argc, char* argv[])
+{
+	if (argc == 1)
+	{
+		std::cout << "No command line arguments " << std::endl;
+		return 1;
+	}
+	std::cout << argv[1] << std::endl;
+	size_t num = std::stoul(argv[1]);
```
Comment (dimagerasimov) [inline TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp:12]: What if I pass a word to your program?

Comment (dimagerasimov) [reply TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp:12]: Regarding to your question, you highly likely crashed with std::out_of_range at this point. It forced me to rewrite this line to cause an std::bad_alloc in my hardware:
`auto num = std::stoull(argv[1]);`

**Review by dimagerasimov — CHANGES_REQUESTED** [review]
You keep doing well.

File: TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp (hunk 1 above)
Comment (dimagerasimov) [reply TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp]: please handle the exception if your input is not a number.

Comment (TimurKM) [reply TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp]: Done

File: TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	std::cout << argv[1] << std::endl;
+
+	unsigned long long num;
+	try
+	{
+		num = std::stoull(argv[1]);
+	}
+	catch (const std::invalid_argument& e)
+	{
+		std::cout << "Allocation failed words: " << e.what() << std::endl;
```
Comment (dimagerasimov) [inline TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp:20]: Just a minor thing. The message should be "Parse failed: ".

File: TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	}
+	catch (const std::bad_alloc& e)
+	{
+		std::cout << "Allocation failed: " << e.what() << std::endl;
+		return -1;
+	}
+
+	std::cout << p << std::endl;
+	delete[] p;
+	return 0;
```
Comment (dimagerasimov) [inline TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp:37]: If the pointer implies re-using, then the next line after deletion should be p = nullptr;. Just keep it in mind.

**Review by dimagerasimov — APPROVED** [review]
Approve. Yet there's one minor comment. You can apply if you wish or you can merge.

File: TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp (hunk 2 above)
Comment (TimurKM) [reply TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp]: Fixed

File: TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp (hunk 3 above)
Comment (TimurKM) [reply TaskWithMemoryAllocation/TaskWithMemoryAllocation.cpp:38]: Fixed

**Pull request merged**